src/main/java/com/jlr/ttl/ds/api/JwtAuthorisationFilter.java

```diff
@@ -0,0 +1,134 @@
+package com.jlr.ttl.ds.api;
+
+import io.jsonwebtoken.Claims;
+import io.jsonwebtoken.JwtException;
+import io.jsonwebtoken.Jwts;
+import jakarta.servlet.FilterChain;
+import jakarta.servlet.ServletException;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.http.HttpStatus;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.security.core.authority.SimpleGrantedAuthority;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.util.StringUtils;
+import org.springframework.web.filter.OncePerRequestFilter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.LinkedHashMap;
+import java.util.List;
+
+/**
+ * @author PCORREI1
+ *<p>
+ * Configuration class that runs whenever there is an API HTTP request. This runs for every GET/POST/PUT/DELETE request. The filter is applied to determine if the user has permission to access the request and the response is tailored to reflect RBAC.
```

Comment:
Do not write long single line comments. Please use line breaks to make it readable and organised